### PR TITLE
fix: Add root Suspense component

### PR DESCRIPTION
**Pull request by olerichter00**
The type of this PR is: **Bugfix**

### Description

Because we mix Relay query renderers with Relay hooks, it can happen that components using `useFragment` suspend without being wrapped in a `Suspense` component. This PR adds a root `Suspense` component to avoid them from throwing an error. 

**Example:**

```
<MyQueryRenderer>
  ...
    <NestedComponentUsingRelayHooks />
  ...
</MyQueryRenderer>
```

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Add root Suspense component - ole

</details>

### CODE REVIEW

**Review by pvinis — APPROVED** [review]
very nice

File: src/lib/AppProviders.tsx [hunk 1]
```diff
@@ -25,6 +25,7 @@ export const AppProviders = ({ children }: { children?: ReactNode }) =>
       RelayDefaultEnvProvider,
       ThemeProvider, // uses: GlobalStoreProvider
       RetryErrorBoundary,
+      SuspenseProvider,
```
Comment (MounirDhahri) [inline src/lib/AppProviders.tsx:28]: **thought:** would it make sense to maybe move this a bit more down in the hierarchy to be able to use it inside the FancyModal context?

Comment (olerichter00) [reply src/lib/AppProviders.tsx:28]: @pvinis `SuspenseProvider` should wrap `_FancyModalPageWrapper` if it is above it, right?

Comment (MounirDhahri) [reply src/lib/AppProviders.tsx:28]: I thought of this in the opposite way where we wrap the `SuspenseProvider` with  `_FancyModalPageWrapper` to stay withing the fancy modal context but it's probably not something we do need for now but we better keep it in mind

Comment (MounirDhahri) [reply src/lib/AppProviders.tsx:28]: this is probably not needed anyway because we have suspense wrappers on the component level as well 😄

Comment (olerichter00) [reply src/lib/AppProviders.tsx:28]: 👍

**Pull request merged**